gnomad_qc/v4/resources/basics.py

```diff
@@ -48,26 +48,61 @@ def get_gnomad_v4_vds(
         vmt = hl.experimental.sparse_split_multi(vmt, filter_changed_loci=True)
         vds = hl.vds.VariantDataset(vds.reference_data, vmt)
 
+    # Count current number of samples in the VDS
+    n_samples = vds.variant_data.count_cols()
+
     # Obtain withdrawn UKBB samples
     excluded_ukbb_samples_ht = hl.import_table(
-        "gs://gnomad/v4.0/ukbb/w26041_20200820.csv"
-    )
+        "gs://gnomad/v4.0/ukbb/w26041_20200820.csv", no_header=True,
+    ).key_by("f0")
 
-    # Count number of samples that should be excluded in the UKBB ht
-    excluded_ukbb_samples = hl.literal(
-        excluded_ukbb_samples_ht.aggregate(
-            hl.agg.collect_as_set(excluded_ukbb_samples_ht.f0)
+    sample_map_ht = hl.read_table("gs://gnomad/v4.0/ukbb/array_sample_map_freeze_7.ht")
+    sample_map_ht = sample_map_ht.annotate(
+        withdrawn_consent=hl.is_defined(
+            excluded_ukbb_samples_ht[sample_map_ht.ukbb_app_26041_id]
         )
     )
-    logger.info(
-        "Total number of UKBB samples to exclude: %d",
-        hl.eval(hl.len(excluded_ukbb_samples)),
-    )
-    # Count current number of samples in the VDS
-    n_samples = vds.variant_data.count_cols()
+    withdrawn_ids = sample_map_ht.filter(sample_map_ht.withdrawn_consent).s.collect()
+
+    # Remove samples that are known to be on the pharma's sample remove list
+    # See https://github.com/broadinstitute/ukbb_qc/blob/70c4268ab32e9efa948fe72f3887e1b81d8acb46/ukbb_qc/resources/basics.py#L308
+    dups_ht = hl.read_table("gs://gnomad/v4.0/ukbb/pharma_known_dups_7.ht")
+
+    ids_to_remove = dups_ht.aggregate(hl.agg.collect(dups_ht["Sample Name - ID1"]))
+
+    # Remove 27 fully duplicated IDs (same exact name for 's' in the VDS)
+    # See https://github.com/broadinstitute/ukbb_qc/blob/70c4268ab32e9efa948fe72f3887e1b81d8acb46/ukbb_qc/resources/basics.py#L286
```

Comment:
Can you confirm with @ch-kr that this will work? I know she needed to do something with indexing (https://github.com/broadinstitute/ukbb_qc/blob/freeze_7/ukbb_qc/utils/prep_dup_map.py) to make sure that we used the newest sample, and I think we need to do that as well to make sure our sample QC matches the samples that we keep. We might need to do the same thing, or see if all the samples are in the same order in the VDS with a possible shift in that actual index because of the merge with gnomAD v4 samples

Reply:
What's the path to the MT that was used for the VDS?

Reply:
Oh, actually, I don't think we have that anymore, can you find out from @ch-kr if there is a HT (or text file) that would have the same original ordering of samples before removing any from the raw 455K MT?